platform/test/spec/testToolManagerSpec.js

```diff
@@ -1,18 +1,185 @@
+/*global describe, it, expect, spyOn, beforeEach, afterEach --  functions provided by Jasmine */
+/*global jasmine --  object provided by Jasmine */
 
-import {ToolManager} from "../../src/ToolsManager.js"
-
+import { ToolManager } from "../../src/ToolsManager.js";
+import { TOOL_1PANELDEF_1FUNCTION } from "../resources/TestToolFiles.js";
+import "jasmine-ajax"
 
 describe("ToolManager", () => {
 
+    describe("constructor", () => {
         it("can be created", () => {
-        
-                var tm = new ToolManager(["test-url"]);
-                
-                var tmType = typeof tm;
-                
-                expect(tmType).toBe("object");
-        
-        })
-        
+            // Call the target object
+            let tm = new ToolManager();
+
+            // Check the expected results
+            expect(tm).toBeInstanceOf(ToolManager);
+        })
+    })
+
+    describe("initialisation", () => {
+         // Setup
+        const TOOL_URLS= [ "test://t1.url/tool-config.json" ];
+        let tm;
+
+        beforeEach( () => {
+            jasmine.Ajax.install();
+           
+            spyOn(XMLHttpRequest.prototype, "open").and.callThrough();
+            spyOn(XMLHttpRequest.prototype, "send").and.callThrough();
+
+            tm = new ToolManager();
+        })
+
+        afterEach(function() {
+            jasmine.Ajax.uninstall();
+        });
+         
+        it("tools URLs can be set", () => {
+            // Call the target object
+            tm.setToolsUrls(TOOL_URLS);
+
+            // Check the expected results
+            for (const t in tm.toolsUrls){
+                expect(tm.toolsUrls[t].url).toEqual(TOOL_URLS[t]);
+            }
```

Comment:
Should there be a check that all URLs have been added?

Reply:
Revised the test to test to check all URLs input and included multiple URLs in the input. Fixed in 0f8cd9c894cfa9fc709b55c649f8af02bfd66cf5